recipes/full_finetune_distributed.py

```diff
@@ -266,6 +267,22 @@ def _setup_model(
         if self._dtype == torch.bfloat16:
             model = model.to(torch.bfloat16)
 
+        # Activation checkpointing ['none', 'full', 'selective']
+        ac_enabled = cfg_file.get("enable_activation_checkpointing", "False")
+
+        # we can't guarantee selective ac is present in the config...
+        ac_mode = cfg_file.get("ac_mode", None)
+        ac_option = cfg_file.get("ac_option", None)
+
+        print(f"ac_enabled: {ac_enabled}, ac_mode: {ac_mode}, ac_option: {ac_option}")
+
+        if ac_enabled and ac_mode:
+            apply_selective_activation_checkpointing(
```

Comment:
when we first did AC for FSDP, it used to require being done after but in theory it should not have mattered.  Regardless, there was an error at the time with autograd engine unless done after...anyway, the old way was to do it after but now it can be done up front.  
Rohan might have more info on this as I don't know the more detailed underlying reasons from back then re: autograd error.